.github/workflows/continuous_deployment.yml

```diff
@@ -182,74 +191,49 @@ jobs:
-
-    - uses: google-github-actions/setup-gcloud@master
-      name: Authenticate with gcloud 🎫
-      with:
-        version: "${{ env.GCLOUD_VERSION }}"
-        service_account_email: ${{ env.SARA_GKE_SERVICE_ACCOUNT_NAME }}
-        service_account_key: ${{ secrets.SARA_GKE_SERVICE_ACCOUNT_KEY }}
+        # Fetch the latest model based on when it created
+        MODEL_URL=$(gsutil ls -l ${{ secrets.STORAGE_BUCKET_URL }}/rasa_demo_models/ | grep "tar.gz" | grep $RASA_VERSION | sort -k2 | tail -n1 | awk '{print $3}')
+        # Get the model local path, get rid of bucket rul
+        MODEL_PATH=$(basename $MODEL_URL)
+        # Download the model from bucket
+        gsutil -m cp ${MODEL_URL} ${MODEL_PATH}
+
+        echo "MODEL_PATH=${MODEL_PATH}" >> $GITHUB_ENV
+        echo "MODEL_NAME=$(basename $MODEL_PATH .tar.gz)" >> $GITHUB_ENV
 
-    - name: Upload model to storage bucket
-      run: |
-          gsutil cp "${MODEL_PATH}" ${STORAGE_BUCKET_URL}/rasa_demo_models
 
     - name: Upload model to Rasa X
       working-directory: ${{ github.workspace }}
+      timeout-minutes: 10
       run: |
-           auth_token=$(curl --request POST \
-              --url https://${RASA_X_DOMAIN}/api/auth \
-              --header 'Content-Type: application/json' \
-              --data "{
-              \"username\": \"${RASA_X_USERNAME}\",
-              \"password\": \"${RASA_X_PASSWORD}\"
-            }" | jq -r '.access_token')
-
-            echo "AUTH_TOKEN=${auth_token}" >> $GITHUB_ENV
-
-            curl -k -F "model=@${MODEL_PATH}" \
-              --url https://$RASA_X_DOMAIN/api/projects/default/models \
-              --header "Authorization: Bearer ${auth_token}"
+          auth_token=$(curl -s -X POST \
+            -H 'Accept: application/json' \
+            -H 'Content-Type: application/json' \
+            --data '{\"username\": \"${RASA_X_USERNAME}\",\"password\": \"${RASA_X_PASSWORD}\"}' \
+            https://${{ env.RASA_X_DOMAIN }}/api/auth | tee /tmp/output_auth.txt | jq -r '.access_token')
+
+          echo "AUTH_TOKEN=${auth_token}" >> $GITHUB_ENV
+
+          # Upload model
+           curl -X POST \
+            -H "Authorization: Bearer ${auth_token}" \
+            -F "model=@${{ env.MODEL_PATH }}" \
+            https://${{ env.RASA_X_DOMAIN }}/api/projects/default/models
+
+          # wait util the model is ready
+          while [[ $(curl -s -X GET -H "Authorization: Bearer ${auth_token}" -H 'Accept: appliration/json' -H 'Content-Type: application/json' https://${{ env.RASA_X_DOMAIN }}/api/projects/default/models | tee /tmp/output.txt | jq -r '.[]? | select(.model=="${{ env.MODEL_NAME }}") | .model') != "${{ env.MODEL_NAME }}" ]]; do
+            # need to wait until the model is available before we can tag it
+            echo "Waiting for model ${MODEL_PATH} to be available.."
+            cat /tmp/output.txt
+            echo ""
+            sleep 10
+          done
```

Comment:
Yes, this is the falling part. This step upload model to Rasa X instance and check the health of the uploaded model. The code is borrowed from [here](https://github.com/RasaHQ/rasa-x/blob/785c2780abd5140ceb6c8e58d93413dd73b53121/.github/workflows/automatic-k8s-deployer.yml#L408-L415). It is working there. I can also test the code locally. I remember we added it because the Rasa X needs some time to prepare the model so we can tag it. 

I also tried to exclude this 'wait for model' part and the [results](https://github.com/RasaHQ/rasa-demo/runs/3534460153) are the same (```***"reasons":["Auth required."],"exception":"Unauthorized"***```), plus false positive ([step passed, but model was not there.](https://github.com/RasaHQ/rasa-demo/runs/3534460153#step:14:40)). 

I am thinking maybe we can pull this problem out of the PR? The model uploading problem in CD is beyond the purpose of this PR and it is better to investigate and solve it on a new PR. What do you think?

Reply:
Isn't the upload itself the purpose of the PR? I don't think we should merge this if it means future PR's will get merged without uploading a new model to Sara.

Reply:
Is this part working again now?

Reply:
Nope. It is still not working.

Reply:
Will this https://github.com/RasaHQ/rasa-demo/pull/652#issuecomment-919225311 fix it?